keras_hub/src/models/retinanet/non_max_supression.py

```diff
@@ -0,0 +1,586 @@
+# Copyright 2024 The KerasHub Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import math
+
+import keras
+from keras import ops
+
+from keras_hub.src.bounding_box import converters
+from keras_hub.src.bounding_box import utils
+from keras_hub.src.bounding_box import validate_format
+
+EPSILON = 1e-8
+
+
+class NonMaxSuppression(keras.layers.Layer):
+    """A Keras layer that decodes predictions of an object detection model.
+
+    Args:
+        bounding_box_format: The format of bounding boxes of input dataset.
+            Refer
+            [to the keras.io docs](https://keras.io/api/keras_cv/bounding_box/formats/)
+            for more details on supported bounding box formats.
+        from_logits: boolean, True means input score is logits, False means
+            confidence.
+        iou_threshold: a float value in the range [0, 1] representing the
+            minimum IoU threshold for two boxes to be considered
+            same for suppression. Defaults to 0.5.
+        confidence_threshold: a float value in the range [0, 1]. All boxes with
+            confidence below this value will be discarded, defaults to 0.5.
+        max_detections: the maximum detections to consider after nms is applied.
+            A large number may trigger significant memory overhead,
+            defaults to 100.
+    """  # noqa: E501
+
+    def __init__(
+        self,
+        bounding_box_format,
+        from_logits,
+        iou_threshold=0.5,
+        confidence_threshold=0.5,
+        max_detections=100,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self.bounding_box_format = bounding_box_format
+        self.from_logits = from_logits
+        self.iou_threshold = iou_threshold
+        self.confidence_threshold = confidence_threshold
+        self.max_detections = max_detections
+        self.built = True
+
+    def call(
+        self, box_prediction, class_prediction, images=None, image_shape=None
+    ):
+        """Accepts images and raw predictions, and returns bounding box
```

Comment:
make this short enough to fit on one line

Reply:
Done